recurring_contract/models/contract_group.py

```diff
@@ -80,17 +77,11 @@ def write(self, vals):
             Perform various check at contract modifications
             - Advance billing increased or decrease
             - Recurring value or unit changes
-            - Another change method was selected
         """
         res = True
         for group in self:
-            # Get the method to apply changes
-            change_method = vals.get('change_method', group.change_method)
-            change_method = getattr(group, change_method)
-
             res = super(ContractGroup, group).write(vals) & res
-            change_method()
-
+            group._updt_invoices_cg(vals)
         return res
```

Comment:
I don't think we are obliged to loop over the records. In my opinion working on the whole set directly will improve performance:
```
res = super().write(vals)
self._updt_invoices_cg(vals)
return res
```